delocate/delocating.py

```diff
@@ -578,6 +592,274 @@ def _make_install_name_ids_unique(
         validate_signature(lib)
 
 
+def _get_macos_min_version(dylib_path: Path) -> Iterator[tuple[str, Version]]:
+    """Get the minimum macOS version from a dylib file.
+
+    Parameters
+    ----------
+    dylib_path : Path
+        The path to the dylib file.
+
+    Yields
+    ------
+    str
+        The CPU type.
+    Version
+        The minimum macOS version.
+    """
+    if dylib_path.is_dir() or not _is_macho_file(dylib_path):
+        return
+    m = MachO(dylib_path)
+    for header in m.headers:
+        for cmd in header.commands:
+            if cmd[0].cmd == LC_BUILD_VERSION:
+                version = cmd[1].minos
+            elif cmd[0].cmd == LC_VERSION_MIN_MACOSX:
+                version = cmd[1].version
+            else:
+                continue
+            yield (
+                CPU_TYPE_NAMES.get(header.header.cputype, "unknown"),
+                Version(f"{version >> 16 & 0xFF}.{version >> 8 & 0xFF}"),
+            )
+            break
+
+
+def _get_archs_and_version_from_wheel_name(
+    wheel_name: str,
+) -> dict[str, Version]:
+    """
+    Get the architecture and minimum macOS version from the wheel name.
+
+    Parameters
+    ----------
+    wheel_name : str
+        The name of the wheel.
+
+    Returns
+    -------
+    Dict[str, Version]
+        A dictionary containing the architecture and minimum macOS version
+        for each architecture in the wheel name.
+    """
+    platform_tag_set = parse_wheel_filename(wheel_name)[-1]
+    res = {}
+    for platform_tag in platform_tag_set:
+        match = _PLATFORM_REGEXP.match(platform_tag.platform)
+        if match is None:
+            raise ValueError(f"Invalid platform tag: {platform_tag.platform}")
+        major, minor, arch = match.groups()
+        res[arch] = Version(f"{major}.{minor}")
+    return res
+
+
+def _get_problematic_libs(
+    version: Optional[Version], version_lib_dict: Dict[Version, List[Path]]
+) -> set[tuple[Path, Version]]:
+    """
+    Filter libraries that require more modern macOS
+    version than the provided one.
+
+    Parameters
+    ----------
+    version : Version or None
+        The expected minimum macOS version.
+        If None, return an empty set.
+    version_lib_dict : Dict[Version, List[Path]]
+        A dictionary containing mapping from minimum macOS version to libraries
+        that require that version.
+
+    Returns
+    -------
+    set[tuple[Path, Version]]
+        A set of libraries that require a more modern macOS version than the
+        provided one.
+    """
+    if version is None:
+        return set()
+    result: set[tuple[Path, Version]] = set()
+    for v, lib in version_lib_dict.items():
+        if v > version:
+            result.update((x, v) for x in lib)
+    return result
```

Comment:
Names should be more verbose. `lib` was actually multiple libraries which was not implied by its name.
```suggestion
def _get_problematic_libs(
    required_version: Optional[Version], version_lib_dict: Dict[Version, List[Path]]
) -> set[tuple[Path, Version]]:
    """
    Filter libraries that require more modern macOS
    version than the provided one.

    Parameters
    ----------
    required_version : Version or None
        The expected minimum macOS version.
        If None, return an empty set.
    version_lib_dict : Dict[Version, List[Path]]
        A dictionary containing mapping from minimum macOS version to libraries
        that require that version.

    Returns
    -------
    set[tuple[Path, Version]]
        A set of libraries that require a more modern macOS version than the
        provided one.
    """
    if required_version is None:
        return set()
    bad_libraries: set[tuple[Path, Version]] = set()
    for library_version, libraries in version_lib_dict.items():
        if library_version > required_version:
            bad_libraries.update((path, library_version) for path in libraries)
    return bad_libraries
```